server/src/main/java/org/cloudfoundry/identity/uaa/provider/JdbcIdentityProviderProvisioning.java

```diff
@@ -218,15 +219,21 @@ public IdentityProvider mapRow(ResultSet rs, int rowNum) throws SQLException {
                 switch (identityProvider.getType()) {
                     case OriginKeys.SAML:
                         definition = JsonUtils.readValue(config, SamlIdentityProviderDefinition.class);
-                        Optional.ofNullable(definition).map(SamlIdentityProviderDefinition.class::cast).ifPresent(e -> e.setIdpEntityId(externId));
+                        if (isNotEmpty(externId)) {
+                            Optional.ofNullable(definition).map(SamlIdentityProviderDefinition.class::cast).ifPresent(e -> e.setIdpEntityId(externId));
```

Comment:
Instead of the surrounding `if` statement, we could also use `.filter(UaaStringUtils::isNotEmpty)` before the call to `ifPresent`.

For me, both is fine.

Reply:
no this is not working since the externalKey is not part of the config JSON....